src/display/canvas.js

```diff
@@ -1188,6 +1188,16 @@ class CanvasGraphics {
       1
     );
 
+    if (width / widthScale <= 5 || height / heightScale <= 5) {
```

Comment:
Should we perhaps limit this to cases where *both* dimensions are small, hence replace the OR with AND above?